### PR TITLE
feat: add swagger config , install package : swagger-jsdoc, swagger-u…

### PR DESCRIPTION
**Pull request by feritka79**
add swagger

### CODE REVIEW

**Comment by feritka79** [comment]
pleas check title and description .

**Review by feritka79 — COMMENTED** [review]
Fix

**Review by feritka79 — COMMENTED** [review]
add remove token api

File: src/routes/swagger/apis/token.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+ *           application/json:
+ *             schema:
+ *                 type: object
+ *                 properties:
+ *                   status:
+ *                     type: string
+ *                     example: error
+ *                   message:
+ *                     type: string
+ *                     example: Token Invalid ...
```
Comment (ceydae) [inline src/routes/swagger/apis/token.ts:106]: change content to: "Failed to save the token"

File: src/routes/swagger/apis/token.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+ *
+ * components:
+ *
+ *     securitySchemes:
+ *       Authorization:
+ *         type: apiKey
+ *         name: authorization
+ *         in: header
+ *
+ *
```
Comment (ceydae) [inline src/routes/swagger/apis/token.ts:121]: Add a authorization field besides the body fields for this route.

File: src/routes/swagger/apis/tokenMint.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
+/**
+ * @swagger
+ * /token/mint:
+ *   post:
+ *     summary: Mint token
```
Comment (ceydae) [inline src/routes/swagger/apis/tokenMint.ts:5]: Mints a new token and transfers the tokens to the specified destination address.

File: src/routes/swagger/apis/tokenMint.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+ *     tags: [Apis]
```
Comment (ceydae) [inline src/routes/swagger/apis/tokenMint.ts:6]: change to "token"

File: src/routes/swagger/apis/tokenMint.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+ *       required: true
+ *       content:
+ *         application/json:
+ *           schema:
+ *             type: object
+ *             properties:
+ *               token:
+ *                 type: string
+ *                 required: true
+ *                 example : GBLBJBTC2URCWUTIXY42W7M5GAZ2NIKTS4QF77BHHWHSBKKSPS2DTOHA
```
Comment (ceydae) [inline src/routes/swagger/apis/tokenMint.ts:17]: change to contract-like address, starting with C

File: src/routes/swagger/apis/tokenMint.ts [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+ *     responses:
+ *       200:
+ *         description:  Minted token
```
Comment (ceydae) [inline src/routes/swagger/apis/tokenMint.ts:20]: Token has been minted successfully

File: src/routes/swagger/apis/tokenMint.ts [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+ *                     _id: "6538cee619e2c76fe6cad499",
+ *                     "address": "CASS3CUNR7W4ASUCEGOMK3TUWITT7KKDS6DQ2TS27UPKRAAKTSHHUJPB",
+ *                     symbol: fDAI,
+ *                     name: FakeDAI,
+ *                     decimals: 7,
+ *                     __v: 0
+ *                   },]
+ *
+ *       400:
+ *         description: User minted
```
Comment (ceydae) [inline src/routes/swagger/apis/tokenMint.ts:44]: User has already minted the fake tokens

File: src/routes/swagger/apis/tokenMint.ts [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+ *                     example: {}
+ * components:
+ *
+ *     securitySchemes:
+ *       Authorization:
+ *         type: apiKey
+ *         name: authorization
+ *         in: header
+ *
+ *
```
Comment (ceydae) [inline src/routes/swagger/apis/tokenMint.ts:85]: no need for authorization for this route.

File: src/routes/swagger/apis/tokenMint.ts
Comment (ceydae) [inline src/routes/swagger/apis/tokenMint.ts:1]: move this file to /src/swagger/token/mint.ts

File: src/routes/swagger/apis/tokenDelete.ts
Comment (ceydae) [inline src/routes/swagger/apis/tokenDelete.ts:1]: move this file to /src/swagger/token/delete.ts

File: src/routes/swagger/apis/tokenDelete.ts [hunk 9]
```diff
@@ -0,0 +1,122 @@
+/**
+ * @swagger
+ * /token:
+ *   delete:
+ *     summary: Delete token
```
Comment (ceydae) [inline src/routes/swagger/apis/tokenDelete.ts:5]: Deletes a token from the list of claimable tokens.

File: src/routes/swagger/apis/tokenDelete.ts [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+ *     tags: [Apis]
```
Comment (ceydae) [inline src/routes/swagger/apis/tokenDelete.ts:6]: change tag to token

File: src/routes/swagger/apis/tokenDelete.ts [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+ *       required: true
+ *       content:
+ *         application/json:
+ *           schema:
+ *             type: object
+ *             properties:
+ *               token:
+ *                 type: string
+ *                 required: true
+ *                 example : GBLBJBTC2URCWUTIXY42W7M5GAZ2NIKTS4QF77BHHWHSBKKSPS2DTOHA
```
Comment (ceydae) [inline src/routes/swagger/apis/tokenDelete.ts:17]: chang example

File: src/routes/swagger/apis/tokenDelete.ts [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+ *     responses:
+ *       200:
+ *         description: Deleted token
```
Comment (ceydae) [inline src/routes/swagger/apis/tokenDelete.ts:20]: Token has been deleted successfully

File: src/routes/swagger/apis/tokenDelete.ts [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 status:
+ *                   type: string
+ *                   example: success
+ *                 message:
+ *                   type: string
+ *                   example: Deleted token successful
```
Comment (ceydae) [inline src/routes/swagger/apis/tokenDelete.ts:31]: change server response and this content to "Token deleted successfully"

File: src/routes/swagger/apis/tokenDelete.ts [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+ *                     _id: 6539a4b453971c96d697f9c0,
+ *                     address: CBBDKFZZPWJQADUXHS3CCIXYRYVKK2SOPIOUDNA5SWXRC7B7APZN3I3H,
+ *                     symbol: fDAI,
+ *                     name: FakeDAI,
+ *                     decimals: 7,
+ *                     __v: 0
+ *                   }
+ *
+ *       400:
+ *         description: Token not exist
```
Comment (ceydae) [inline src/routes/swagger/apis/tokenDelete.ts:44]: Token does not exist on the database

File: src/routes/swagger/apis/tokenDelete.ts [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 status:
+ *                   type: string
+ *                   example: error
+ *                 message:
+ *                   type: string
+ *                   example: Token not exists
```
Comment (ceydae) [inline src/routes/swagger/apis/tokenDelete.ts:55]: Change server response and this content to "Token does not exist"

File: src/routes/swagger/apis/tokenDelete.ts [hunk 16]
```diff
@@ -0,0 +1,122 @@
...
+ *                   example: {
+ *                     _id: 6539a4b453971c96d697f9c0,
+ *                     address: CBBDKFZZPWJQADUXHS3CCIXYRYVKK2SOPIOUDNA5SWXRC7B7APZN3I3H,
+ *                     symbol: fDAI,
+ *                     name: FakeDAI,
+ *                     decimals: 7,
+ *                     __v: 0
+ *                   }
+ *
+ *       400:
```
Comment (ceydae) [inline src/routes/swagger/apis/tokenDelete.ts:43]: change the server response's status code to 404

File: src/routes/swagger/apis/tokenDelete.ts [hunk 17]
```diff
@@ -0,0 +1,122 @@
...
+ *                   example: error
+ *                 message:
+ *                   type: string
+ *                   example: Token not exists
+ *                 result:
+ *                   type: object
+ *                   example: {}
+ *
+ *       401:
+ *         description: Authorization
```
Comment (ceydae) [inline src/routes/swagger/apis/tokenDelete.ts:61]: change to Authorization failed

File: src/routes/swagger/apis/tokenDelete.ts [hunk 18]
```diff
@@ -0,0 +1,122 @@
...
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 status:
+ *                   type: string
+ *                   example: error
+ *                 message:
+ *                   type: string
+ *                   example: Authorization not found
```
Comment (ceydae) [inline src/routes/swagger/apis/tokenDelete.ts:72]: Authorization failed

File: src/routes/swagger/apis/tokenDelete.ts [hunk 19]
```diff
@@ -0,0 +1,122 @@
...
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 status:
+ *                   type: string
+ *                   example: error
+ *                 message:
+ *                   type: string
+ *                   example: Access not found
```
Comment (ceydae) [inline src/routes/swagger/apis/tokenDelete.ts:89]: Authorization failed

File: src/routes/swagger/apis/tokenDelete.ts [hunk 20]
```diff
@@ -0,0 +1,122 @@
...
+ *
+ *
+ * components:
+ *
+ *     securitySchemes:
+ *       Authorization:
+ *         type: apiKey
+ *         name: authorization
+ *         in: header
+ *
```
Comment (ceydae) [inline src/routes/swagger/apis/tokenDelete.ts:120]: add a header field for authorization for this route.

**Review by ceydae — CHANGES_REQUESTED** [review]

**Review by feritka79 — COMMENTED** [review]
fix

File: src/utils/token/removeToken.ts
Comment (ceydae) [inline src/utils/token/removeToken.ts:1]: This file is not needed. Remove it

File: src/utils/token/removeToken.ts [hunk 21]
```diff
@@ -0,0 +1,23 @@
...
+      responseTemplate({
+        status: 'error',
+        message: e.message,
+        result: {},
+      });
+    }
+  }
+};
+
+export default removeToken;
```
Comment (ceydae) [inline src/utils/token/removeToken.ts:23]: Remove the whole file

**Review by ceydae — CHANGES_REQUESTED** [review]

**Review by ceydae — APPROVED** [review]

**Pull request merged**